test/plugin/scenarios/shenyu-2.4.x-scenario/config/expectedData.yaml

```diff
@@ -0,0 +1,80 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+segmentItems:
+  - serviceName: shenyu-http
```

Comment:
You should consider exposing this as test entry, and there is a new endpoint calling Gateway, and Gateway routes back to here.

Then we could verify whether Gateway process incoming context coreectly.

Reply:
The current calling relationship is as follows:
client http request -> ShenYu gateway（route to provider） -> shenyu-http(service provider)